doc/src/guides/contributing/deprecations.md

```diff
@@ -0,0 +1,388 @@
+(deprecation-policy)=
+# Deprecation Policy
+
+This page outlines SymPy's police on doing deprecations, and describes the
+steps developers should take to properly deprecate code.
+
+## What is a deprecation?
+
+A deprecation is a way to make backwards incompatible changes in a way that
+allows users to update their code. Deprecated code continues to work as it
+used to, but whenever someone uses it, it prints a warning to the screen
+indicating that it will be removed in a future version of SymPy, and
+indicating what the user should be using instead.
+
+This allows users a chance to update their code without it completely
+breaking. It also gives SymPy an opportunity to give users an informative
+message on how to update their code, rather than making their code simply
+error or start giving wrong answers.
+
+## When to deprecate and when not to
+
+Sometimes the API of SymPy must change in an incompatible way. We try to avoid
+this, because changing API breaks people's code, but it is often deemed worthy
```

Comment:
but it can be deemed worthy to do so in some circumstances.

Reply:
I think our default position should be to discourage API changes unless the benefit largely outweighs the cost. Of course, whether something largely outweighs the cost is subjective because we aren't measuring any information to make it objective. But I think the language in this document should give the overall impression that you should make a very strong effort in avoiding API changes at all and make a very strong effort to design APIs that have less likelihood of needing to be changed. Opening the document with language that sets a strong tone early is good. If our language is weak at the opening about this, then the actual actions of developers will be even weaker (maybe too much API changes tolerated).